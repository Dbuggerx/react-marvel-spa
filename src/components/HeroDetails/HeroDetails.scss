.hero-details {
  $list-margin: 10px;
  font-size: 1.5em;

  &__image {
    margin: 15px auto;
    display: block;
  }

  &__lists {
    display: flex;
    flex-flow: row wrap;
    justify-content: space-between;
    margin-left: -$list-margin;
  }

  &__list {
    flex: 1 1 auto;
    margin-left: $list-margin;
    margin-top: $list-margin;
  }
}