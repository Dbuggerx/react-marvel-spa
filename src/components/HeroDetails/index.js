// @flow

import React from 'react';
import type { StatelessFunctionalComponent } from 'react';
import NameList from '../NameList';
import type { Hero } from '../../services/types';
import './HeroDetails.scss';

type Props = {
    hero: Hero,
    imageUrl: string,
};

/**
 * Renders a collection of string
 */
const renderListFor = (
    collectionName: string,
    collection: {
        items: $Shape<{
            resourceURI: string,
            name: string,
            type: string,
        }>[],
    },
) =>
    (collection && collection.items.length > 0 ? (
        <div className="hero-details__list">
            <NameList collectionName={collectionName} names={collection.items.map(c => c.name)} />
        </div>
    ) : null);

/**
 * Renders the image and detailed information for the Hero
 */
const HeroDetails: StatelessFunctionalComponent<Props> = (props: Props) => (
    <div className="hero-details">
        <img className="hero-details__image" src={props.imageUrl} alt={props.hero.name} />
        {props.hero.description.length > 0 && props.hero.description}
        <div className="hero-details__lists">
            {renderListFor('Comics', props.hero.comics)}
            {renderListFor('Series', props.hero.series)}
            {renderListFor('Stories', props.hero.stories)}
            {renderListFor('Events', props.hero.events)}
        </div>
    </div>
);

HeroDetails.displayName = 'HeroDetails';

export default HeroDetails;
