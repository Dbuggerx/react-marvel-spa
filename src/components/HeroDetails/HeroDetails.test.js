import React from 'react';
import { shallow } from 'enzyme';
import HeroDetails from './index';

const basicHeroMock = {
    id: 1011334,
    name: '3-D Man',
    description: 'This is a test!',
    modified: '2014-04-29T14:18:17-0400',
    thumbnail: {
        path: 'http://i.annihil.us/u/prod/marvel/i/mg/c/e0/535fecbbb9784',
        extension: 'jpg',
    },
};

describe('HeroDetails', () => {
    test('renders basic data', () => {
        const wrapper = shallow(<HeroDetails hero={basicHeroMock} imageUrl="the/img/url" />);
        expect(wrapper).toMatchSnapshot();
    });

    test('renders collections data', () => {
        const mock = {
            ...basicHeroMock,
            comics: {
                items: [
                    {
                        name: 'comicA',
                    },
                    {
                        name: 'comicB',
                    },
                ],
                returned: 0,
            },
            series: {
                items: [
                    {
                        name: 'serieA',
                    },
                    {
                        name: 'serieB',
                    },
                ],
                returned: 0,
            },
        };
        const wrapper = shallow(<HeroDetails hero={mock} imageUrl="the/img/url" />);
        expect(wrapper).toMatchSnapshot();
    });
});
