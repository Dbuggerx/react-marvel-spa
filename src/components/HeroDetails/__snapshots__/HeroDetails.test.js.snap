// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroDetails renders basic data 1`] = `
<div
  className="hero-details"
>
  <img
    alt="3-D Man"
    className="hero-details__image"
    src="the/img/url"
  />
  This is a test!
  <div
    className="hero-details__lists"
  />
</div>
`;

exports[`HeroDetails renders collections data 1`] = `
<div
  className="hero-details"
>
  <img
    alt="3-D Man"
    className="hero-details__image"
    src="the/img/url"
  />
  This is a test!
  <div
    className="hero-details__lists"
  >
    <div
      className="hero-details__list"
    >
      <NameList
        collectionName="Comics"
        names={
          Array [
            "comicA",
            "comicB",
          ]
        }
      />
    </div>
    <div
      className="hero-details__list"
    >
      <NameList
        collectionName="Series"
        names={
          Array [
            "serieA",
            "serieB",
          ]
        }
      />
    </div>
  </div>
</div>
`;
